src/daft-core/src/series/ops/is_in.rs

```diff
@@ -0,0 +1,142 @@
+use std::collections::HashSet;
+
+use common_error::{DaftError, DaftResult};
+
+use crate::{datatypes::BooleanArray, IntoSeries, Series};
+
+macro_rules! do_is_in {
+    ($name:expr, $data:expr, $items:expr) => {{
+        let mut set = HashSet::new();
+        for i in 0..$items.len() {
+            set.insert($items.get(i));
+        }
+
+        let mut result = Vec::new();
+        for i in 0..$data.len() {
+            result.push(set.contains(&$data.get(i)));
+        }
+
+        Ok(BooleanArray::from(($name, result.as_slice())).into_series())
+    }};
+}
+
+impl Series {
+    pub fn is_in(&self, items: &Self) -> DaftResult<Series> {
+        let items = &items.list()?.flat_child;
+        let default =
+            BooleanArray::from((self.name(), vec![false; self.len()].as_slice())).into_series();
+        if items.is_empty() {
+            return Ok(default);
+        }
+        // match items.data_type() with the datatypes of LiteralValue because items is a List(Vec<LiteralValue>),
+        // attept to cast self to the same datatype as items, then check if self is in items
+        match items.data_type() {
+            crate::datatypes::DataType::Null => self.is_null(),
```

Comment:
nit: you can do a `use crate::datatypes::DataType` on the top of the file to make these references to `DataType` less verbose